modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -433,7 +437,36 @@ fn generate_ephemeral_tweak(static_pk: PublicKey) -> ([u8; 32], PublicKey) {
 }
 
 impl LightningClientModule {
-    async fn select_gateway(&self) -> Result<(SafeUrl, RoutingInfo), SelectGatewayError> {
+    /// This method updates the mapping from lightning node public keys to
+    /// gateway api endpoints maintained in the module database once a day. When
+    /// paying an invoice this enables the client to select the gateway that has
+    /// created the invoice, if possible, such that the payment does not go
+    /// over lightning, reducing fees and latency.
+    ///
+    /// Client integrators are expected to call this function in a spawned task.
+    pub async fn update_gateway_map(&self) -> ! {
+        loop {
+            if let Ok(gateways) = self.module_api.gateways().await {
```

Comment:
Could use let-else syntax to reduce indentation

```rust
let Ok(gateways) = self.module_api.gateways().await else { continue; };
```

on a second thought though: this should likely retry and never stop retrying till it gets an answer. That way the error case would be completely avoided (assuming we'll never hit 2^64 retries).